<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>FilterWindow</class>
 <widget class="QMainWindow" name="FilterWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>FilterWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QPushButton" name="next_loaded_image_btn">
    <property name="geometry">
     <rect>
      <x>310</x>
      <y>130</y>
      <width>31</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>&gt;</string>
    </property>
   </widget>
   <widget class="QPushButton" name="previous_loaded_image_btn">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>130</y>
      <width>31</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>&lt;</string>
    </property>
   </widget>
   <widget class="Line" name="line">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>440</y>
      <width>741</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>290</y>
      <width>91</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Loaded images:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>490</y>
      <width>71</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Filter name:</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="filter_name_input">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>490</y>
      <width>151</width>
      <height>21</height>
     </rect>
    </property>
   </widget>
   <widget class="QScrollArea" name="dirty_image_area">
    <property name="geometry">
     <rect>
      <x>470</x>
      <y>20</y>
      <width>256</width>
      <height>256</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="widgetResizable">
     <bool>true</bool>
    </property>
    <widget class="QWidget" name="scrollAreaWidgetContents_2">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>254</width>
       <height>254</height>
      </rect>
     </property>
     <widget class="QLabel" name="dirty_image_label">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>110</y>
        <width>111</width>
        <height>16</height>
       </rect>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Ignored" vsizetype="Ignored">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="text">
       <string>No loaded image</string>
      </property>
      <property name="scaledContents">
       <bool>true</bool>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>520</y>
      <width>71</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Filter rank:</string>
    </property>
   </widget>
   <widget class="QTabWidget" name="noise_selection_group">
    <property name="geometry">
     <rect>
      <x>470</x>
      <y>310</y>
      <width>251</width>
      <height>121</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="gauss_noise_tab">
     <attribute name="title">
      <string>Gaussian</string>
     </attribute>
     <widget class="QSlider" name="gaussian_slider">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>40</y>
        <width>160</width>
        <height>16</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QLabel" name="gaussian_value">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>20</y>
        <width>47</width>
        <height>14</height>
       </rect>
      </property>
      <property name="text">
       <string>0</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="snp_noise_tab">
     <attribute name="title">
      <string>Salt and Pepper</string>
     </attribute>
    </widget>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>460</y>
      <width>91</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Filter creation:</string>
    </property>
   </widget>
   <widget class="Line" name="line_2">
    <property name="geometry">
     <rect>
      <x>380</x>
      <y>0</y>
      <width>20</width>
      <height>431</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>470</x>
      <y>290</y>
      <width>91</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Apply noise:</string>
    </property>
   </widget>
   <widget class="QListWidget" name="listWidget_loaded_images">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>310</y>
      <width>251</width>
      <height>121</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>Load images first..</string>
     </property>
     <property name="background">
      <brush brushstyle="NoBrush">
       <color alpha="255">
        <red>171</red>
        <green>171</green>
        <blue>171</blue>
       </color>
      </brush>
     </property>
     <property name="foreground">
      <brush brushstyle="NoBrush">
       <color alpha="255">
        <red>190</red>
        <green>190</green>
        <blue>190</blue>
       </color>
      </brush>
     </property>
    </item>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>310</x>
      <y>500</y>
      <width>146</width>
      <height>22</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QRadioButton" name="rcima_rbtn">
       <property name="text">
        <string>RCIMA</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="fastrcima_rbtn">
       <property name="text">
        <string>fast-RCIMA</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>310</x>
      <y>480</y>
      <width>141</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Filter calculation algorithm:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>470</x>
      <y>0</y>
      <width>91</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Dirty image</string>
    </property>
   </widget>
   <widget class="QPushButton" name="calc_filter_btn">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>510</y>
      <width>80</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>Calculate filter</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>0</y>
      <width>91</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Loaded image</string>
    </property>
   </widget>
   <widget class="QScrollArea" name="loaded_image_area">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>20</y>
      <width>256</width>
      <height>256</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="widgetResizable">
     <bool>true</bool>
    </property>
    <widget class="QWidget" name="scrollAreaWidgetContents">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>254</width>
       <height>254</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QSpinBox" name="rank_value_spinBox">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>520</y>
      <width>42</width>
      <height>22</height>
     </rect>
    </property>
    <property name="acceptDrops">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="load_images_btn">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>280</y>
      <width>80</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>Load images</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_min_max_rank">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>540</y>
      <width>47</width>
      <height>14</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuMenu">
    <property name="title">
     <string>Menu</string>
    </property>
    <addaction name="actionFilter_image"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <addaction name="menuMenu"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionFilter_image">
   <property name="text">
    <string>Filter image</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
  </action>
 </widget>
 <tabstops>
  <tabstop>load_images_btn</tabstop>
  <tabstop>listWidget_loaded_images</tabstop>
  <tabstop>noise_selection_group</tabstop>
  <tabstop>filter_name_input</tabstop>
  <tabstop>rank_value_spinBox</tabstop>
  <tabstop>rcima_rbtn</tabstop>
  <tabstop>fastrcima_rbtn</tabstop>
  <tabstop>calc_filter_btn</tabstop>
  <tabstop>next_loaded_image_btn</tabstop>
  <tabstop>gaussian_slider</tabstop>
  <tabstop>dirty_image_area</tabstop>
  <tabstop>loaded_image_area</tabstop>
  <tabstop>previous_loaded_image_btn</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>gaussian_slider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>gaussian_value</receiver>
   <slot>setNum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>569</x>
     <y>407</y>
    </hint>
    <hint type="destinationlabel">
     <x>593</x>
     <y>384</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
